pytealutils/storage/blob/blob.py

```diff
@@ -0,0 +1,206 @@
+            App.localPut(acct, intkey(i.load()), BytesZero(pageSize))
+        )
+
+    @staticmethod
+    @Subroutine(TealType.uint64)
+    def get_byte(acct: TealType.uint64, idx: TealType.uint64):
+        """
+        Get a single byte from local storage of an account by index
+        """
+        key = intkey(idx / pageSize)
+        offset = idx % pageSize
+        return GetByte(App.localGet(acct, key), offset)
+
+    @staticmethod
+    @Subroutine(TealType.none)
+    def set_byte(acct: TealType.uint64, idx: TealType.uint64, byte: TealType.uint64):
+        """
+        Set a single byte from local storage of an account by index
+        """
+        key = intkey(idx / pageSize)
+        offset = idx % pageSize
+        return App.localPut(acct, key, SetByte(App.localGet(acct, key), offset, byte))
+
+    @staticmethod
+    @Subroutine(TealType.bytes)
+    def read(
+        acct: TealType.uint64, bstart: TealType.uint64, bstop: TealType.uint64
+    ) -> Expr:
+        """
+        read bytes between bstart and bstop from local storage of an account by index
+        """
+
+        startKey = bstart / pageSize
+        startOffset = bstart % pageSize
+
+        stopKey = startKey + (bstop / pageSize)
+        stopOffset = startOffset + (bstop % pageSize)
+
+        key = ScratchVar()
+        buff = ScratchVar()
+
+        start = ScratchVar()
+        stop = ScratchVar()
+
+        init = key.store(startKey)
+        cond = key.load() <= stopKey
+        incr = key.store(key.load() + Int(1))
+
+        return Seq(
+            buff.store(Bytes("")),
+            For(init, cond, incr).Do(
+                Seq(
+                    start.store(If(key.load() == startKey, startOffset, Int(0))),
+                    stop.store(If(key.load() == stopKey, stopOffset, pageSize)),
+                    buff.store(
+                        Concat(
+                            buff.load(),
+                            Substring(
+                                App.localGet(acct, intkey(key.load())),
+                                start.load(),
+                                stop.load(),
+                            ),
+                        )
+                    ),
+                )
+            ),
+            buff.load(),
+        )
+
+    @staticmethod
+    @Subroutine(TealType.uint64)
+    def write(
+        acct: TealType.uint64, bstart: TealType.uint64, buff: TealType.bytes
+    ) -> Expr:
+        """
+        write bytes between bstart and len(buff) to local storage of an account
+        """
+
+        length = Len(buff)
+
+        startKey = bstart / pageSize
+        startOffset = bstart % pageSize
+
+        stopKey = startKey + (length / pageSize)
+        stopOffset = startOffset + (length % pageSize)
+
+        key = ScratchVar()
+        start = ScratchVar()
+        stop = ScratchVar()
+        written = ScratchVar()
+
+        init = key.store(startKey)
+        cond = key.load() <= stopKey
```

Comment:
Feels like an off-by-one error that can be fixed with `key.load() < stopKey` instead of `<=`